lib/cadet/assessments/assessments.ex

```diff
@@ -1413,30 +1414,75 @@ defmodule Cadet.Assessments do
     end
   end
 
-  @spec update_submission_status_and_xp_bonus(Submission.t()) ::
+  @spec update_submission_status(Submission.t()) ::
           {:ok, Submission.t()} | {:error, Ecto.Changeset.t()}
-  defp update_submission_status_and_xp_bonus(submission = %Submission{}) do
-    assessment = submission.assessment
-    assessment_conifg = Repo.get_by(AssessmentConfig, id: assessment.config_id)
+  defp update_submission_status(submission = %Submission{}) do
+    submission
+    |> Submission.changeset(%{status: :submitted, submitted_at: Timex.now()})
+    |> Repo.update()
+  end
 
-    max_bonus_xp = assessment_conifg.early_submission_xp
-    early_hours = assessment_conifg.hours_before_early_xp_decay
+  @spec update_xp_bonus(Submission.t()) ::
+          {:ok, Submission.t()} | {:error, Ecto.Changeset.t()}
+  defp update_xp_bonus(submission = %Submission{id: submission_id}) do
+    # to ensure backwards compatibility
```

Comment:
the reason this was here was because I was worried that if a grader went to grade an old submission where submitted_at is null, the xp_bonus will be set to 0 and decrease. Therefore if xp_bonus is not 0 it does nothing and keeps it at whatever value it was previously set to.

Reply:
but, if you have set the `submitted_at`, time, this should not be an worry.

The way to reason this is that think about this as a function, the bonus_xp should be a pure mathematical function of (`submitted_at`, `xp`, `xp_adjustment`).

Otherwise, if the autograder thinks that it is not low effort and bonus_xp is awarded, then the our staff grader will not be able to revoke this low effort 'bonus_xp'.
In the example that I gave here, bonus_xp depends onthe order of operation. Explicitly, if the TA did the adjustment _before_ the bonus xp is awarded, then the bonus xp can be 0.
If the TA did the adjustment _after_ the bonus xp is awarded, then the bonus xp cannot be 0.

If we decided that the TA should not affect the bonus xp behaviour, then we should not trigger update in update grading info.